packages/web3-utils/src/converters.ts

```diff
@@ -255,6 +252,29 @@ export const toNumber = (value: Numbers): number | bigint => {
 	return hexToNumber(numberToHex(value));
 };
 
+/**
+ * Auto converts any given value into it's bigint representation
+ */
+export const toBigInt = (value: unknown): bigint => {
+	if (typeof value === 'number') {
+		return BigInt(value);
+	}
+
+	if (typeof value === 'bigint') {
+		return value;
+	}
+
+	if (typeof value === 'string' && !isHexStrict(value)) {
+		return BigInt(value);
+	}
+
+	if (typeof value === 'string' && isHexStrict(value)) {
+		return BigInt(hexToNumber(value));
+	}
```

Comment:
Instead of using `isHexStrict` here, could we use `isNumber` and combine both `if` statements?

Also `return BigInt(hexToNumber(value));` could result in a over/underflow, so it would be nice to convert number and hex string straight to BigInt without converting to `number`

Reply:
`isNumber` does not work with hex strings.

Reply:
`hexToNumber` actually checks if:

```typescript
if (num > Number.MAX_SAFE_INTEGER) {
		return negative ? -num : num;
	}
```

so if we update it to also check for `Number.MIN_SAFE_INTEGER`, than my concern should be covered